components/director/internal/repo/exist.go

```diff
@@ -0,0 +1,38 @@
+package repo
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/pkg/errors"
+)
+
+type ExistQuerier struct {
+	tableName    string
+	idColumn     string
+	tenantColumn string
+}
+
+func NewExistQuerier(tableName, tenantColumn, idColumn string) *ExistQuerier {
+	return &ExistQuerier{tableName: tableName, idColumn: idColumn, tenantColumn: tenantColumn}
+}
+
+func (g *ExistQuerier) Exists(ctx context.Context, tenant, id string) (bool, error) {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	q := fmt.Sprintf("SELECT 1 FROM %s WHERE %s=$1 AND %s=$2", g.tableName, g.tenantColumn, g.idColumn)
```

Comment:
what if there are different number of conditions than two? I'm concerned that this approach will be too restrictive. I'm still a fan of variadic parameter approach for parameters.